src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html

```diff
@@ -11,6 +11,7 @@
             *jhiExtensionPoint="overrideStudentActions; context: { courseId: courseId, exercise: exercise, showResult: showResults && latestRatedResult != undefined }"
             [courseId]="courseId"
             [exercise]="exercise"
+            [studentParticipation]="gradedStudentParticipation"
```

Comment:
`gradedStudentParticipation` is the first `studentParticipation` where the `testRun` flag is not set. IMO some comments explaining that or a different naming is required.
`testRun` strictly relates to Exam test runs, which the name does not imply.

see `Participation.java`

Reply:
This is no longer the case, it is now also used for practice participations in test runs.
Could you adapt this comment and also convert it to a proper java doc, so it's visible when you hover somewhere over that variable?
Therefore this graded participation is the participation before the deadline

Reply:
D